src/modules/UI/Wallets/action.js

```diff
@@ -86,54 +85,56 @@ export const upsertWallet = (wallet: AbcCurrencyWallet) => (dispatch: any, getSt
   })
 }
 
-// setEnabledTokens is specifically for enabling them *within the GUI*, not within the core
-export const setEnabledTokens = (walletId: string, enabledTokens: any) => (dispatch: any, getState: any) => {
+// adds to core and enables in core
+export const addCustomToken = (walletId: string, tokenObj: any) => (dispatch: any, getState: any) => {
+  const state = getState()
+  const wallet = CORE_SELECTORS.getWallet(state, walletId)
+  WALLET_API.addCoreCustomToken(wallet, tokenObj)
+  .then(() => {
+    return
+  })
+  .catch((e) => console.log(e))
+}
+
+export const setEnabledTokens = (walletId: string, enabledTokens: Array<string>, disabledTokens: Array<string>) => (dispatch: any, getState: any) => {
+  // tell Redux that we are updating the enabledTokens list
   dispatch(setTokensStart())
```

Comment:
Seems like it's overkill to use redux for this. Same for setTokensSuccess(). The spinner is state that's specific to the component and isn't global to the app. Not to be fixed in this PR since it's not a showstopper, but please clean up after this PR is merged